src/device/mod.rs

```diff
@@ -437,6 +437,9 @@ impl Device {
         if let Some(rt) = self.rt.take() {
             if let Some(art) = &self.art {
                 let _ = art.block_on(rt.stop());
+                if self.features.flush_events_on_stop {
+                    while !self.event.is_empty() {}
+                }
```

Comment:
This is a spinloop which will burn CPU cycles, assuming some events might take a while, this is not the best idea. As an easy fix I suggest adding in some sleep in the loop or using some higher level primitive

Reply:
Fixed. I set the sleep to 100ms to keep it from blocking for too long